XLS-39d-clawback/README.md

```diff
@@ -0,0 +1,178 @@
+----------------------- -------------------------------------------------------
+
+### Advantages and Disadvantages
+
+**Advantages**
+
+- Issuers that are restricted from issuing on ledger because of regulatory requirements requiring the ability to clawback funds will be able to issue tokens.
+- By being able to "clawback" issuers can ensure that the "on-chain" view is representative of the balance.
+- Compared to other on-ledger features (e.g. freeze), clawback is minimal and trivial to implement.
+- `Clawback` is disabled by default, this gives token holders more confidence that they won't be clawed back randomly. 
+
+
+**Disadvantages**
+
+- Introduces an additional transaction, along with the complexity that comes with that.
+- Issuers now get additional power, which may be concerning to token holders.
+- Requires additional documentation, including highlighting the fact that clawback cannot be applied to XRP.
+
+---
+
+## 2. Motivation
+Jurisdictions may require issuers of digital assets to have a way to recover funds in certain circumstances. The `Clawback` feature can provide a way to comply with these regulations.
+
+
+
+---
+
+## 3. Specification
+
+### 3.1. On-Ledger Data Structures
+
+This proposal introduces no new on ledger structures.
+
+
+
+### 3.2. Account Root modifications
+
+This proposal introduces 1 additional flag for the `Flags` field of `AccountRoot`:
+
+| Flag Name       |  Flag Value  |
+|:---------------:|:------------:|
+| `lsfAllowClawback` | `0x02000000` | 
+
+Clawback is disabled by default. The account must set this flag through an `AccountSet` transaction, which is successful only if the account has an empty owner directory, meaning they have no trustlines, offers, escrows, payment channels, or checks. Otherwise, the `AccountSet` returns `tecHAS_OBLIGATIONS`. After this flag has been successfully set, it cannot reverted, and the account permanently gains the ability to clawback on frozen trustlines.
+
+If the account attempts to set `lsfAllowClawback` while `lsfNoFreeze` is set, the transaction will return `tecNO_PERMISSION` because clawback cannot be enabled on an account that has already disclaimed the ability to freeze trustlines. Reversely, if an account attempts to set `lsfNoFreeze` while `lsfAllowClawback` is set, the transaction will also return `tecNO_PERMISSION`.
+
+
+### 3.3. Transactions
+This proposal introduces one new transaction: `Clawback`
+
+#### 3.3.1. `Clawback` transaction
+The **`Clawback`** transaction modifies a trustline object, by adjusting the balance accordingly and, if instructed to, by changing relevant flags. If possible (i.e. if the `Clawback` transaction would leave the trustline is in the "default" state), the transaction will also remove the trustline.
+
+**An counterparty can be clawed back if and only if `lsfAllowClawback` is set.** If this transaction is attempted while `lsfAllowClawback` is unset, it will return with an error code `tecNO_PERMISSION`.
```

Comment:
```suggestion
**Issued tokens can be clawed back by an issuer if, and only if, `lsfAllowClawback` is set.** If this transaction is attempted while `lsfAllowClawback` is unset, it will return with an error code `tecNO_PERMISSION`.
```

Reply:
will change. thanks!